src/scripts/components/main/content-navigation/use-content-event-handlers.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  Dispatch,
+  SetStateAction,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+
+export const useContentTouchHandlers = (
+  currentIndex: number,
+  setCurrentIndex: Dispatch<SetStateAction<number>>,
+  numOfItems: number, // Ensures we stay within bounds
+) => {
+  const [touchStartY, setTouchStartY] = useState<number | null>(null);
+  // Save the current index value in a ref to prevent it from resetting
+  const currentIndexRef = useRef(currentIndex);
+
+  // Update the ref whenever currentIndex changes
+  useEffect(() => {
+    currentIndexRef.current = currentIndex;
+  }, [currentIndex]);
+
+  const handleTouchMove = (e: React.TouchEvent) => {
+    if (touchStartY === null) {
+      setTouchStartY(e.touches[0].clientY);
+      return;
+    }
+
+    const touchDelta = e.touches[0].clientY - touchStartY;
+
+    // For content navigation, max should be number of items minus 1
+    const maxIndex = Math.floor(numOfItems / 2);
+    const minIdex = Math.floor((numOfItems - 1) / 2) * -1;
+
+    const itemHeight = 32; // Height of each item in pixels
+    const sensitivity = 2; // Adjust this to control movement sensitivity
```

Comment:
Constants should be written in uppercase and be placed at the top of the script. Please use descriptive names